### PR TITLE
doc: add release notes to standalone doc bundle

### PR DESCRIPTION
**Pull request by notriddle**
Preview: http://notriddle.com/rustdoc-html-demo-5/release-notes/releases.html

This is a workaround for #101714 on top of being a useful addition in its own right. It is intended to change the "canonical URL" for viewing the release notes from GitHub, which is relatively slow, to a pre-rendered HTML file that loads from the same CDN as the standard library docs. It also means you get a copy of the release notes when installing the rust-docs with rustup.

### CODE REVIEW

File: src/bootstrap/src/core/build_steps/doc.rs [hunk 1]
```diff
@@ -388,6 +389,106 @@ impl Step for Standalone {
...
+        let html = out.join("releases.html");
+        let tmppath = out.join("releases.md");
+        let inpath = builder.src.join("RELEASES.md");
+        let rustdoc = builder.rustdoc(compiler);
+        if !up_to_date(&inpath, &html)
+            || !up_to_date(&footer, &html)
+            || !up_to_date(&favicon, &html)
+            || !up_to_date(&full_toc, &html)
+            || !(builder.config.dry_run() || up_to_date(&version_info, &html))
+            || !(builder.config.dry_run() || up_to_date(&rustdoc, &html))
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/build_steps/doc.rs:449]: No need to check `dry_run` twice

```suggestion
            || !(builder.config.dry_run() || up_to_date(&version_info, &html) || up_to_date(&rustdoc, &html))
```

**Review by onur-ozkan — APPROVED** [review]
Nice workaround, LGTM!

r=me with the nit suggestion

**Comment by onur-ozkan** [comment]
While I like it very much, I don't want to make the decision alone, so I would like one more person to review it.

r? Mark-Simulacrum

File: src/bootstrap/src/core/build_steps/doc.rs [hunk 2]
```diff
@@ -388,6 +389,107 @@ impl Step for Standalone {
...
+    }
+
+    /// Generates all standalone documentation as compiled by the rustdoc in `stage`
+    /// for the `target` into `out`.
+    ///
+    /// This will list all of `src/doc` looking for markdown files and appropriately
+    /// perform transformations like substituting `VERSION`, `SHORT_HASH`, and
+    /// `STAMP` along with providing the various header/footer HTML we've customized.
+    ///
+    /// In the end, this is just a glorified wrapper around rustdoc!
```
Comment (Mark-Simulacrum) [inline src/bootstrap/src/core/build_steps/doc.rs:421]: This comment looks potentially outdated to me?

**Comment by Mark-Simulacrum** [comment]
r=me beyond the small comment

**Pull request merged**